sdks/python/apache_beam/testing/analyzers/perf_regression_analysis.py

```diff
@@ -0,0 +1,372 @@
+    """
+    Args:
+     pvalue: p value for the permutation test.
+     permutations: Number of permutations for the permutation test.
+     For more information, please look at
+     https://pypi.org/project/signal-processing-algorithms/
+
+    Performs edivisive means on the data and returns the indices of the
+    Change points.
+
+    Returns:
+     The indices of change points.
+    """
+    return e_divisive(self.data, pvalue, permutations)
+
+
+def is_change_point_in_valid_window(
+    num_runs_in_change_point_window: int, change_point_index: int) -> bool:
+  # If the change point is more than N runs behind the most recent run,
+  # Ignore the change point and don't raise an alert for it.
+  return num_runs_in_change_point_window >= change_point_index
+
+
+def find_existing_issue(
+    metric_name: str,
+    test_name: str,
+    change_point_timestamp: pd.Timestamp,
+    sibling_change_point_min_timestamp: pd.Timestamp,
+    sibling_change_point_max_timestamp: pd.Timestamp,
+) -> Optional[Tuple[bool, Optional[int]]]:
+  """
+  Finds the most recent GitHub issue created for change points for this
+  test+metric in sibling change point min and max timestamps window.
+  Returns a boolean and an issue ID whether the issue needs to be updated.
+  """
+  query_template = f"""
+  SELECT * FROM {_BQ_PROJECT_NAME}.{_BQ_DATASET}.{test_name}
+  WHERE {METRIC_NAME} = '{metric_name}'
+  ORDER BY {ISSUE_CREATION_TIMESTAMP_LABEL} DESC
+  LIMIT 1
+  """
+  try:
+    df = BigQueryMetricsFetcher().get_metrics(
+        query_template=query_template,
+        limit=_NUM_DATA_POINTS_TO_RUN_CHANGE_POINT_ANALYSIS)
+  except exceptions.NotFound:
+    # If no table found, that means this is first performance regression
+    # on the current test+metric.
+    return True, None
+  issue_number = df[ISSUE_NUMBER].tolist()[0]
+
+  if (sibling_change_point_min_timestamp <= change_point_timestamp <=
+      sibling_change_point_max_timestamp):
+    return False, None
+  return True, issue_number
+
+
+def read_test_config(config_file_path: str) -> Dict:
+  """
+  Reads the config file in which the data required to
+  run the change point analysis is specified.
+  """
+  with open(config_file_path, 'r') as stream:
+    config = yaml.safe_load(stream)
+  return config
+
+
+def run(config_file_path: str = None) -> None:
+  """
+  run is the entry point to run change point analysis on test metric
+  data, which is read from config file, and if there is a performance
+  regression observed for a test, an alert will filed with GitHub Issues.
+
+  If config_file_path is None, then the run method will use default
+  config file to read the required perf test parameters.
+
+  """
+  if config_file_path is None:
+    config_file_path = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), 'tests_config.yaml')
+
+  tests_config: Dict[Dict[str, Any]] = read_test_config(config_file_path)
+
+  # min_runs_between_change_points, num_runs_in_change_point_window can be
```

Comment:
the comment seems out of place. you move add the reference to README in the docstring for `run` and remove the rest.